File: frontend/src/services/adminService.js
/**
 * adminService.js — Admin API Service
 *
 * This file should contain:
 * - Import apiClient from './apiClient'
 * - Functions:
 *     - getAllUsers(filters) → GET /admin/users?role=&status=
 *     - getUserById(id) → GET /admin/users/{id}
 *     - suspendUser(userId, reason) → POST /admin/users/{id}/suspend
 *     - unsuspendUser(userId) → POST /admin/users/{id}/unsuspend
 *     - getPlatformStats() → GET /admin/stats
 *
 * Purpose:
 *   Centralizes all admin API calls.
 *   Used by AdminDashboard, UserManagementPage.
 *
 * Export: { getAllUsers, getUserById, suspendUser, unsuspendUser, getPlatformStats }
 */

File: backend/WedaLK/demo/src/main/java/lk/wedalk/admin/service/AdminService.java
package lk.wedalk.admin.service;

/**
 * AdminService.java — Admin Business Logic
 *
 * This file should contain:
 * - @Service annotation
 * - Inject UserRepository, ServiceRequestRepository, DisputeRepository,
 * VerificationRepository
 * - Methods:
 * - List<UserDto> getAllUsers(Role filterByRole)
 * - void suspendUser(Long userId, SuspendUserRequest request)
 * - void unsuspendUser(Long userId)
 * - Map<String, Object> getPlatformStats()
 * - Total users (by role)
 * - Total requests (by status)
 * - Total open disputes
 * - Total pending verifications
 *
 * Purpose:
 * Admin-specific business logic — user management and platform statistics.
 */
